scheduler/src/cook/mesos/api.clj

```diff
@@ -170,13 +170,28 @@
    (s/optional-key :cancelled) s/Bool
    (s/optional-key :reason_string) s/Str})
 
+(defn max-128-characters-and-alphanum?
+  "Returns true if s contains only '.', '_', '-' or
+  any word characters and has length at most 128"
+  [s]
+  (re-matches #"[\.a-zA-Z0-9_-]{0,128}" s))
+
+(defn non-empty-max-128-characters-and-alphanum?
+  "Returns true if s contains only '.', '_', '-' or any
+  word characters and has length at least 1 and most 128"
+  [s]
+  (re-matches #"[\.a-zA-Z0-9_-]{1,128}" s))
+
+(def Application
+  "Schema for the application a job corresponds to"
+  {:name (s/constrained s/Str non-empty-max-128-characters-and-alphanum?)
+   :version (s/constrained s/Str non-empty-max-128-characters-and-alphanum?)})
+
 (def Job
   "A schema for a job"
   {:uuid s/Uuid
    :command s/Str
-   ;; Make sure the job name is a valid string which can only contain '.', '_', '-' or any word characters and has
-   ;; length at most 128.
-   :name (s/both s/Str (s/pred #(re-matches #"[\.a-zA-Z0-9_-]{0,128}" %) 'under-128-characters-and-alphanum?))
+   :name (s/both s/Str (s/pred max-128-characters-and-alphanum? 'max-128-characters-and-alphanum?))
```

Comment:
Agreed, but it seems like we shouldn't break people who are sending empty string names